domain/src/main/java/org/oppia/android/domain/platformparameter/PlatformParameterSingleton.kt

```diff
@@ -0,0 +1,53 @@
+package org.oppia.android.domain.platformparameter
+
+import org.oppia.android.util.platformparameter.PlatformParameter
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+
+/** Singleton class which helps in storing and providing Platform Parameters at runtime. */
+@Singleton
+class PlatformParameterSingleton @Inject constructor() {
+  private var platformParameterMap: Map<String, ParameterValue> = mapOf()
+
+  /** Get the current [platformParameterMap]. */
+  fun getPlatformParameterMap() = platformParameterMap
+
+  /** Initialize [platformParameterMap] if it is not yet initialised. */
+  fun setPlatformParameterMap(map: Map<String, ParameterValue>) {
+    if (platformParameterMap.isEmpty()) platformParameterMap = map
+  }
+
+  /** Retrieve a String type [ParameterValue], if it exists in the [platformParameterMap]. */
```

Comment:
Retrieve --> Retrieves. Add `@param` and `@return` annotations

Reply:
Done